scripts/prepare_heat_demand.py

```diff
@@ -2,15 +2,439 @@
+
+    Returns
+    -------
+    files_region : list
+        List of file names of region
+    """
+    files_region = [file for file in os.listdir(path) if region in file]
+    files_region = sorted(files_region)
+
+    if not files_region:
+        raise FileNotFoundError(
+            f"No data of region {region} could be found in directory: {path}."
+        )
+
+    return files_region
+
+
+def get_year(file_name):
+    """
+    This function returns a year from file name
+
+    Parameters
+    ----------
+    file_name : str
+        Name of file with year in it
+
+    Returns
+    -------
+    year : int
+        Year
+    """
+    # Add array with years to be searched for in file name
+    years_search_array = np.arange(1990, 2051)
+    newline = "\n"
+
+    year_in_file = [
+        year_searched
+        for year_searched in years_search_array
+        if str(year_searched) in file_name
+    ]
+    if len(year_in_file) == 1:
+        year = year_in_file[0]
+    else:
+        raise ValueError(
+            f"Your file {file_name} is missing a year or has multiple years "
+            f"in its name." + newline + "Please provide data for a single year "
+            "with that year in the file name."
+        )
+
+    return year
+
+
+def get_holidays(path, year, region):
+    """
+    This function determines all holidays of a given region in a given year
+
+    Parameters
+    ----------
+    path : str
+        Input path
+    year : int
+        Year
+    Returns
+    -------
+    holidays_dict : dict
+        Dictionary with holidays
+
+    """
+    # Read all national holidays per state
+    all_holidays = pd.read_csv(path)
+    holidays_dict = {}
+
+    # Get holidays in region
+    holidays_filtered = all_holidays.loc[all_holidays["year"] == year]
+    holidays_filtered = holidays_filtered[
+        holidays_filtered["region"].str.contains(region)
+    ]
+
+    for row in holidays_filtered.iterrows():
+        holidays_dict[
+            datetime.date(row[1]["year"], row[1]["month"], row[1]["day"])
+        ] = row[1]["holiday"]
+
+    return holidays_dict
+
+
+def get_building_class(region, path):
+    """
+    This function reads building classes of German states from input path
+    and returns the building class of given region
+
+    Parameters
+    ----------
+    path : str
+        Input path
+    region : str
+        Region (eg. Brandenburg)
+
+    Returns
+    -------
+    building_class : str
+         Building class (German: Baualtersklasse) can assume values in range 1-11
+         eg. 5 in case of BB and 3 in case of B
+    """
+    # Read all national holidays per state
+    all_building_classes = pd.read_csv(path)
+    building_class_df = all_building_classes.loc[
+        all_building_classes["region"] == region
+    ]
+    building_class = building_class_df["building_class"].values[0]
+
+    return building_class
+
+
+def check_central_decentral(demands, value, consumer, carrier):
```

Comment:
This function is hard to understand.

Reply:
I hope my changes in commit https://github.com/rl-institut/oemof-B3/pull/76/commits/77aabd9316ba8f2a602203f13af6e1740c4f4454 help.